File: C++/level 5/2 x n tiling.cpp
#include<iostream>
#include<vector>
using namespace std;

int tiling(int n)
{
    unsigned int answer = 0;
    vector<unsigned int> arr;
    arr.push_back(1);
    arr.push_back(1);
  
    if(n==1)
        answer = arr[n-1];
    else{
  	    for(int i=2; i<n+1; i++){
    	    arr.push_back((arr[i-2]+arr[i-1])%100000);
        }
        answer = arr[n];
    }
	return answer;
}
int main()
{
	int testn = 110;
	int testAnswer = tiling(testn);

	cout<< testAnswer;
}

File: Javascript/level 5/2 x n tiling.js
function tiling(n) {
	var answer = 0;
    var arr = [1, 1]
    
    if(n==1)
        answer = arr[0];
    else{
        for(var i=2; i<n+1; i++){
            arr.push((arr[i-2]+arr[i-1])%100000);
        }
        answer = arr[n];
    }
	return answer;
}

// 아래는 테스트로 출력해 보기 위한 코드입니다.
console.log(tiling(2));